dumper/dataWriter.go

```diff
@@ -179,7 +184,7 @@ func substitutePrimaryKey(table schemareader.Table, row []sqlUtil.RowDataStructu
 		pkSequence = true
 	}
 	for _, column := range row {
-		if pkSequence && strings.Compare(column.ColumnName, "id") == 0 {
+		if pkSequence && table.PKColumns[column.ColumnName] {
```

Comment:
It can work but I'm not sure this is so simple. If we want to check with the `table.PKColumns` in this way, then I think we also should check that the PKColumns only contains one element.